src/types.ts

```diff
@@ -1,6 +1,13 @@
-export type Cat = Readonly<{
-  id: string;
-  name: string;
+export type PagePaginatedData<ResponseDataType> = Readonly<{
+  results: ReadonlyArray<ResponseDataType>;
```

Comment:
[explanation] This construct of `<T>` is called generics (and is actually not limited to only TypeScript, but also in Java) 
https://www.typescriptlang.org/docs/handbook/2/generics.html
As can be read in the documentation above, this will allow us to be flexible in what we pass, but w/o losing any info on typing we would have had if we would have used `any`

```typescript
function identity(arg: any): any {
  return arg;
}
```

> While using `any` is certainly generic in that it will cause the function to accept any and all types for the type of `arg`, we actually are losing the information about what that type was when the function returns. If we passed in a number, the only information we have is that any type could be returned.